Odata-docs/webapi-8/fundamentals/overview.md

```diff
@@ -0,0 +1,582 @@
+        }
+    ]
+}
+```
+
+ASP.NET Core OData transforms the query options into an [ODataQueryOptions](/dotnet/api/microsoft.aspnetcore.odata.query.odataqueryoptions) object that converts the query options into [LINQ](/dotnet/csharp/programming-guide/concepts/linq/) expressions. `[EnableQuery]` uses the `ODataQueryOptions` object to transform the result from the controller action. If the result is an [`IEnumerable<T>`](/dotnet/api/system.collections.generic.ienumerable-1) collection (for example an in-memory `List<T>`), the data filtering and transformations will be performed in-memory. If the result is an [`IQueryable<T>`](/dotnet/api/system.linq.iqueryable-1) collection, the query options are transformed into [LINQ expression trees](/dotnet/csharp/programming-guide/concepts/expression-trees/) that are passed to the underlying [query provider](/dotnet/api/system.linq.iqueryprovider). The underlying provider can translate the expression trees into native database queries that are executed directly on the database.
+
+## Request and response format
+
+ASP.NET Core OData uses JSON format for its requests and responses (the `$metadata` endpoint returns the schema in XML format by default). The OData JSON format may include metadata in the payload such as type information that makes it possible for clients and server to know how to handle the payload. This metadata is usually expressed in the form of "annotations" which are extra fields in the payload that start with `@odata.` prefix.
+
+For example. Let's say we have an OData service that has a `Customers` entity set which returns entities of type `Customer`.
+
+For a request like `GET /Customers`, we would a get a response like
+
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Customers",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1"
+        },
+        {
+            "Id": 2,
+            "Name": "Customer 2"
+        },
+        {
+            "Id": 3,
+            "Name": "Customer 3"
+        }
+    ]
+}
+```
+
+Notice that this response does not just return an array of `Customer` object, it returns an object comprising of `@odata.context` and `value` properties. The `value` property's value is the array that contains the actual data. The `@odata.context` property is an annotation that gives that client information about where this data comes from. The `http://localhost:5000/odata/$metadata#Customers` is called the context URL, in this case it shows that this payload comes from the `Customers` entity set. The client can find out more information about the `Customers` entity set from the schema, which it can be retrieved from the service metadata endpoint - `http://localhost:5000/$metadata`. Using both the schema and the context URL, the client can tell that the payload value is a an array of `Customer` objects, and it can determine the properties contained in the `Customer` entity type and their respective types.
```

Comment:
```suggestion
Notice that this response does not just return an array of `Customer` object, it returns an object comprising of `@odata.context` and `value` properties. The `value` property's value is the array that contains the actual data. The `@odata.context` property is an annotation that gives that client information about where this data comes from. The `http://localhost:5000/$metadata#Customers` is called the context URL, in this case it shows that this payload comes from the `Customers` entity set. The client can find out more information about the `Customers` entity set from the schema, which can be retrieved from the service metadata endpoint - `http://localhost:5000/$metadata`. Using both the schema and the context URL, the client can tell that the payload value is a an array of `Customer` objects, and it can determine the properties contained in the `Customer` entity type and their respective types.
```